topics/request.md

```diff
@@ -1,316 +1,116 @@
-For example:
+### HTTP method {id="http-method"}
 
-### Sequential requests
+When calling the `request` function, you can specify the desired HTTP method using the `method` property:
 
 ```kotlin
-suspend fun sequentialRequests() {
-    val client = HttpClient()
-
-    // Get the content of an URL.
-    val firstBytes = client.get<ByteArray>("https://127.0.0.1:8080/a")
-
-    // Once the previous request is done, get the content of an URL.
-    val secondBytes = client.get<ByteArray>("https://127.0.0.1:8080/b")
-
-    client.close()
-}
 ```
+{src="snippets/_misc_client/RequestMethodWithParams.kt"}
 
-### Parallel requests
-
+In addition to the `request` function, `HttpClient` provides specific functions for basic HTTP methods: [get](https://api.ktor.io/%ktor_version%/io.ktor.client.request/get.html), [post](https://api.ktor.io/%ktor_version%/io.ktor.client.request/post.html), [put](https://api.ktor.io/%ktor_version%/io.ktor.client.request/put.html), and so on. For example, you can replace the example above with the following code:
 ```kotlin
-suspend fun parallelRequests() = coroutineScope<Unit> {
-    val client = HttpClient()
-
-    // Start two requests asynchronously.
-    val firstRequest = async { client.get<ByteArray>("https://127.0.0.1:8080/a") }
-    val secondRequest = async { client.get<ByteArray>("https://127.0.0.1:8080/b") }
-
-    // Get the request contents without blocking threads, but suspending the function until both
-    // requests are done.
-    val bytes1 = firstRequest.await() // Suspension point.
-    val bytes2 = secondRequest.await() // Suspension point.
-
-    client.close()
-}
 ```
+{src="snippets/_misc_client/GetMethodWithoutParams.kt"}
 
-## Customizing requests
-
-We cannot live only on *get* requests, Ktor allows you to build complex requests with any of the HTTP verbs, with the flexibility to process responses in many ways.
-
-### Default http methods
-
-{id="shortcut-methods"}
-
-Similar to `request`, there are several extension methods to perform requests
-with the most common HTTP verbs: `GET`, `POST`, `PUT`, `DELETE`, `PATCH`, `HEAD` and `OPTIONS`.
-
+### Headers {id="headers"}
+To add headers to the request, use the [headers](https://api.ktor.io/%ktor_version%/io.ktor.client.request/-http-request-builder/headers.html) function as follows:
 ```kotlin
-val text = client.post<String>("http://127.0.0.1:8080/")
 ```
+{src="snippets/_misc_client/GetMethodWithHeaders.kt"}
 
-When calling request methods, you can provide a lambda to build the request
-parameters like the URL, the HTTP method(verb), the body, or the headers:
 
-```kotlin
-val text = client.post<String>("http://127.0.0.1:8080/") {
-    header("Hello", "World")
-}
-```
 
-The [HttpRequestBuilder](https://api.ktor.io/%ktor_version%/io.ktor.client.request/-http-request-builder/) looks like this:
+### Cookies {id="cookies"}
+To send cookies, use the [cookie](https://api.ktor.io/%ktor_version%/io.ktor.client.request/cookie.html) function:
```

Comment:
We use a lot "as follows". Not sure it's necessary. Here you don't use it, but in above examples you do. Maybe remove?

Reply:
ok, removed and rephased